test/Benchmarks/OrleansBenchmarks/Serialization/SerializationBenchmarks.cs

```diff
@@ -1,58 +1,311 @@
+                this.complexClass,
+                new AnotherConcreteClass
+                {
+                    AnotherString = "hi",
+                    Interfaces = new List<ISomeInterface>
+                    {
+                        this.complexClass
+                    }
+                },
+                new AnotherConcreteClass(),
+                OuterClass.GetPrivateClassInstance()
             };
+            
+            this.complexClass.Classes = classes.ToArray();
+            this.complexClass.Enum = SomeAbstractClass.SomeEnum.Something;
+            this.complexClass.SetObsoleteInt(38);
 
-            SerializationManager.InitializeForTesting();
+            this.complexClass.Struct = new SomeStruct(10)
+            {
+                Id = Guid.NewGuid(),
+                PublicValue = 6,
+                ValueWithPrivateGetter = 7
+            };
+            this.complexClass.Struct.SetValueWithPrivateSetter(8);
+            this.complexClass.Struct.SetPrivateValue(9);
 
-            BufferPool.InitGlobalBufferPool(new MessagingConfiguration(false));
+
+            this.largeTestData = new LargeTestData
+            {
+                Description = "This is a test. This is only a test. In the event of a real execution, this would contain actual data.",
+                EnumValue = TestEnum.First
+            };
+            largeTestData.SetBit(13);
+            largeTestData.SetEnemy(17, CampaignEnemyTestType.Enemy1);
+
+            this.serializedBytes = SerializationManager.SerializeToByteArray(this.largeTestData);
         }
 
         [Benchmark]
-        public void SerializerBenchmark()
+        public byte[] SerializerBenchmark()
         {
-            for (int i = 0; i < Repeats; i++)
+            return SerializationManager.SerializeToByteArray(this.largeTestData);
+        }
+
+        [Benchmark]
+        public object DeserializerBenchmark()
+        {
+            return SerializationManager.DeserializeFromByteArray<LargeTestData>(this.serializedBytes);
+        }
+
+        /// <summary>
+        /// Performs a full serialization loop using a type which has not had code generation performed.
+        /// </summary>
+        /// <returns></returns>
+        [Benchmark]
+        public object FallbackFullLoop()
+        {
+            return OrleansSerializationLoop(this.complexClass);
+        }
+
+        internal static object OrleansSerializationLoop(object input, bool includeWire = true)
+        {
+            var copy = SerializationManager.DeepCopy(input);
+            if (includeWire)
             {
-                var bytes = SerializationManager.SerializeToByteArray(poco);
+                copy = SerializationManager.RoundTripSerializationForTesting(copy);
             }
+            return copy;
         }
+    }
 
-        [Benchmark]
-        public void DeserializerBenchmark()
+    [Serializable]
+    internal struct SomeStruct
```

Comment:
These types are included here and made internal so that they are inaccessible to the code generator. This is in order to force the use of the fallback serializer.